src/ir/debuginfo.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_ir_debuginfo_h
+#define wasm_ir_debuginfo_h
+
+#include "wasm.h"
+
+namespace wasm::debuginfo {
+
+// Given an expression and another that it replaces, copy the debug info from
+// the latter to the former. Note that the expression may not be an exclusive
+// replacement of the other (the other may be replaced by several expressions,
+// all of whom may end up with the same debug info).
+inline void copyDebugInfoToReplacement(Expression* replacement,
+                                       Expression* original,
```

Comment:
Nit: How about putting the `original` first? Source first and then the destination seems more intuitive to me, but if you like the current order it's fine too.

Reply:
Good point, yeah, this was not that clear. I reversed the order now and also renamed it so it is totally unambiguous, `copyOriginalToReplacement`. I also removed `DebugInfo` from the name as they are always called as `debuginfo::copyFoo()` anyhow, so it is clear we are copying debuginfo.